flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/DefaultDeclarativeSlotPool.java

```diff
@@ -599,4 +632,27 @@ private ResourceCounter getFulfilledRequirements(
     ResourceCounter getFulfilledResourceRequirements() {
         return fulfilledResourceRequirements;
     }
+
+    @VisibleForTesting
+    @Nonnull
+    public ComponentMainThreadExecutor getComponentMainThreadExecutor() {
+        return componentMainThreadExecutor;
+    }
+
+    @VisibleForTesting
+    @Nonnull
+    public Duration getSlotRequestMaxInterval() {
+        return slotRequestMaxInterval;
+    }
+
+    @VisibleForTesting
+    public void tryWaitSlotRequestMaxIntervalTimeout() {
```

Comment:
```suggestion
    void tryWaitSlotRequestIsDone() {
```

1. Don't need public.
2. We are waiting for SlotRequestIsDone instead of fixed timeout.